src/zarr/abc/store.py

```diff
@@ -19,7 +20,34 @@
 
 __all__ = ["ByteGetter", "ByteSetter", "Store", "set_or_delete"]
 
-ByteRangeRequest: TypeAlias = tuple[int | None, int | None]
+
+@dataclass
+class ExplicitByteRequest:
```

Comment:
```suggestion
class ExplicitByteRangeRequest:
```
I think we should go for the longer name for consistency.

Reply:
strictly speaking this specifies an interval that doesn't have anything to do with bytes or requests so maybe we can use the name `ExplicitInterval`

Reply:
What about `RangeByteRequest` to match `<Suffix|Offset>ByteRequest`? Specifically, I'm not really sure how much interpretability `Explicit` adds.

Reply:
All of these are fine with me. I was just thinking that the name of the union type should be consistent with the invidiual types.
1. `ByteRangeRequest: TypeAlias = ExplicitByteRangeRequest | OffsetByteRangeRequest | SuffixByteRangeRequest`
2. `ByteRequest: TypeAlias = RangeByteRequest | OffsetByteRequest | SuffixByteRequest`
3. `Interval: TypeAlias = ClosedInterval | LeftOpenInterval | RightOpenInterval`

I prefer 1 or 2 because they are only used for the`byte_range` argument. But no strong opinion.

Reply:
I also prefer 1 or 2 and will chose 2 for now given that it's more concise. I'm fine with other options though if someone else has a strong opinion on the name.